File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/create_git_folder_details.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class CreateGitFolderDetails(object):
    """
    The data to create a Git folder in a workspace.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new CreateGitFolderDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param folder_path:
            The value to assign to the folder_path property of this CreateGitFolderDetails.
        :type folder_path: str

        :param git_provider_key:
            The value to assign to the git_provider_key property of this CreateGitFolderDetails.
        :type git_provider_key: str

        :param git_repository_url:
            The value to assign to the git_repository_url property of this CreateGitFolderDetails.
        :type git_repository_url: str

        :param branch_name:
            The value to assign to the branch_name property of this CreateGitFolderDetails.
        :type branch_name: str

        :param credential_key:
            The value to assign to the credential_key property of this CreateGitFolderDetails.
        :type credential_key: str

        :param description:
            The value to assign to the description property of this CreateGitFolderDetails.
        :type description: str

        """
        self.swagger_types = {
            'folder_path': 'str',
            'git_provider_key': 'str',
            'git_repository_url': 'str',
            'branch_name': 'str',
            'credential_key': 'str',
            'description': 'str'
        }

        self.attribute_map = {
            'folder_path': 'folderPath',
            'git_provider_key': 'gitProviderKey',
            'git_repository_url': 'gitRepositoryUrl',
            'branch_name': 'branchName',
            'credential_key': 'credentialKey',
            'description': 'description'
        }

        self._folder_path = None
        self._git_provider_key = None
        self._git_repository_url = None
        self._branch_name = None
        self._credential_key = None
        self._description = None

    @property
    def folder_path(self):
        """
        **[Required]** Gets the folder_path of this CreateGitFolderDetails.
        The absolute path of the Git folder user wants to create.


        :return: The folder_path of this CreateGitFolderDetails.
        :rtype: str
        """
        return self._folder_path

    @folder_path.setter
    def folder_path(self, folder_path):
        """
        Sets the folder_path of this CreateGitFolderDetails.
        The absolute path of the Git folder user wants to create.


        :param folder_path: The folder_path of this CreateGitFolderDetails.
        :type: str
        """
        self._folder_path = folder_path

    @property
    def git_provider_key(self):
        """
        Gets the git_provider_key of this CreateGitFolderDetails.
        key corresponding to Git service provider in git provider table.


        :return: The git_provider_key of this CreateGitFolderDetails.
        :rtype: str
        """
        return self._git_provider_key

    @git_provider_key.setter
    def git_provider_key(self, git_provider_key):
        """
        Sets the git_provider_key of this CreateGitFolderDetails.
        key corresponding to Git service provider in git provider table.


        :param git_provider_key: The git_provider_key of this CreateGitFolderDetails.
        :type: str
        """
        self._git_provider_key = git_provider_key

    @property
    def git_repository_url(self):
        """
        **[Required]** Gets the git_repository_url of this CreateGitFolderDetails.
        Git repository url used to clone.


        :return: The git_repository_url of this CreateGitFolderDetails.
        :rtype: str
        """
        return self._git_repository_url

    @git_repository_url.setter
    def git_repository_url(self, git_repository_url):
        """
        Sets the git_repository_url of this CreateGitFolderDetails.
        Git repository url used to clone.


        :param git_repository_url: The git_repository_url of this CreateGitFolderDetails.
        :type: str
        """
        self._git_repository_url = git_repository_url

    @property
    def branch_name(self):
        """
        **[Required]** Gets the branch_name of this CreateGitFolderDetails.
        Git branch name that is cloned.


        :return: The branch_name of this CreateGitFolderDetails.
        :rtype: str
        """
        return self._branch_name

    @branch_name.setter
    def branch_name(self, branch_name):
        """
        Sets the branch_name of this CreateGitFolderDetails.
        Git branch name that is cloned.


        :param branch_name: The branch_name of this CreateGitFolderDetails.
        :type: str
        """
        self._branch_name = branch_name

    @property
    def credential_key(self):
        """
        **[Required]** Gets the credential_key of this CreateGitFolderDetails.
        Credential key of the stored git credentials.


        :return: The credential_key of this CreateGitFolderDetails.
        :rtype: str
        """
        return self._credential_key

    @credential_key.setter
    def credential_key(self, credential_key):
        """
        Sets the credential_key of this CreateGitFolderDetails.
        Credential key of the stored git credentials.


        :param credential_key: The credential_key of this CreateGitFolderDetails.
        :type: str
        """
        self._credential_key = credential_key

    @property
    def description(self):
        """
        Gets the description of this CreateGitFolderDetails.
        Short description about the git repository.


        :return: The description of this CreateGitFolderDetails.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this CreateGitFolderDetails.
        Short description about the git repository.


        :param description: The description of this CreateGitFolderDetails.
        :type: str
        """
        self._description = description

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/CreateGitFolderDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * The data to create a Git folder in a workspace.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=CreateGitFolderDetails.Builder.class)

public final class CreateGitFolderDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"folderPath", "gitProviderKey", "gitRepositoryUrl", "branchName", "credentialKey", "description"})
    public CreateGitFolderDetails(String folderPath, String gitProviderKey, String gitRepositoryUrl, String branchName, String credentialKey, String description) {
        super();
        this.folderPath = folderPath;
        this.gitProviderKey = gitProviderKey;
        this.gitRepositoryUrl = gitRepositoryUrl;
        this.branchName = branchName;
        this.credentialKey = credentialKey;
        this.description = description;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The absolute path of the Git folder user wants to create.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("folderPath")
private String folderPath;

        /**
         * The absolute path of the Git folder user wants to create.
         * @param folderPath the value to set
         * @return this builder
         **/
        

public Builder folderPath(String folderPath) {
    this.folderPath = folderPath;
    return this;
}
            /**
     * key corresponding to Git service provider in git provider table.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitProviderKey")
private String gitProviderKey;

        /**
         * key corresponding to Git service provider in git provider table.
         * @param gitProviderKey the value to set
         * @return this builder
         **/
        

public Builder gitProviderKey(String gitProviderKey) {
    this.gitProviderKey = gitProviderKey;
    return this;
}
            /**
     * Git repository url used to clone.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitRepositoryUrl")
private String gitRepositoryUrl;

        /**
         * Git repository url used to clone.
         * @param gitRepositoryUrl the value to set
         * @return this builder
         **/
        

public Builder gitRepositoryUrl(String gitRepositoryUrl) {
    this.gitRepositoryUrl = gitRepositoryUrl;
    return this;
}
            /**
     * Git branch name that is cloned.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("branchName")
private String branchName;

        /**
         * Git branch name that is cloned.
         * @param branchName the value to set
         * @return this builder
         **/
        

public Builder branchName(String branchName) {
    this.branchName = branchName;
    return this;
}
            /**
     * Credential key of the stored git credentials.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("credentialKey")
private String credentialKey;

        /**
         * Credential key of the stored git credentials.
         * @param credentialKey the value to set
         * @return this builder
         **/
        

public Builder credentialKey(String credentialKey) {
    this.credentialKey = credentialKey;
    return this;
}
            /**
     * Short description about the git repository.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

        /**
         * Short description about the git repository.
         * @param description the value to set
         * @return this builder
         **/
        

public Builder description(String description) {
    this.description = description;
    return this;
}


        public CreateGitFolderDetails build() {
            CreateGitFolderDetails model = new CreateGitFolderDetails(this.folderPath
                , this.gitProviderKey
                , this.gitRepositoryUrl
                , this.branchName
                , this.credentialKey
                , this.description);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(CreateGitFolderDetails model) {
                this.folderPath(model.getFolderPath());
    this.gitProviderKey(model.getGitProviderKey());
    this.gitRepositoryUrl(model.getGitRepositoryUrl());
    this.branchName(model.getBranchName());
    this.credentialKey(model.getCredentialKey());
    this.description(model.getDescription());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The absolute path of the Git folder user wants to create.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("folderPath")
    private final String folderPath;

        /**
     * The absolute path of the Git folder user wants to create.
     * @return the value
     **/
    
    public String getFolderPath() {
        return folderPath;
    }


        /**
     * key corresponding to Git service provider in git provider table.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitProviderKey")
    private final String gitProviderKey;

        /**
     * key corresponding to Git service provider in git provider table.
     * @return the value
     **/
    
    public String getGitProviderKey() {
        return gitProviderKey;
    }


        /**
     * Git repository url used to clone.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitRepositoryUrl")
    private final String gitRepositoryUrl;

        /**
     * Git repository url used to clone.
     * @return the value
     **/
    
    public String getGitRepositoryUrl() {
        return gitRepositoryUrl;
    }


        /**
     * Git branch name that is cloned.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("branchName")
    private final String branchName;

        /**
     * Git branch name that is cloned.
     * @return the value
     **/
    
    public String getBranchName() {
        return branchName;
    }


        /**
     * Credential key of the stored git credentials.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("credentialKey")
    private final String credentialKey;

        /**
     * Credential key of the stored git credentials.
     * @return the value
     **/
    
    public String getCredentialKey() {
        return credentialKey;
    }


        /**
     * Short description about the git repository.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * Short description about the git repository.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("CreateGitFolderDetails(");
        sb.append("folderPath=").append(String.valueOf(this.folderPath));
        sb.append(", gitProviderKey=").append(String.valueOf(this.gitProviderKey));
        sb.append(", gitRepositoryUrl=").append(String.valueOf(this.gitRepositoryUrl));
        sb.append(", branchName=").append(String.valueOf(this.branchName));
        sb.append(", credentialKey=").append(String.valueOf(this.credentialKey));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CreateGitFolderDetails)) {
            return false;
        }

        CreateGitFolderDetails other = (CreateGitFolderDetails) o;
        return java.util.Objects.equals(this.folderPath, other.folderPath) &&
            java.util.Objects.equals(this.gitProviderKey, other.gitProviderKey) &&
            java.util.Objects.equals(this.gitRepositoryUrl, other.gitRepositoryUrl) &&
            java.util.Objects.equals(this.branchName, other.branchName) &&
            java.util.Objects.equals(this.credentialKey, other.credentialKey) &&
            java.util.Objects.equals(this.description, other.description);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.folderPath == null ? 43 : this.folderPath.hashCode());
        result = (result * PRIME) + (this.gitProviderKey == null ? 43 : this.gitProviderKey.hashCode());
        result = (result * PRIME) + (this.gitRepositoryUrl == null ? 43 : this.gitRepositoryUrl.hashCode());
        result = (result * PRIME) + (this.branchName == null ? 43 : this.branchName.hashCode());
        result = (result * PRIME) + (this.credentialKey == null ? 43 : this.credentialKey.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        return result;
    }


}
